generator/src/main/java/com/algorand/sdkutils/utils/TypeDef.java

```diff
@@ -0,0 +1,71 @@
+package com.algorand.sdkutils.utils;
+
+/**
+ * 
+ * TypeDef hold together information about a type
+ * javaTypeName is the generated code type: e.g. List<abc>, or MyEnumClassName
+ * rawTypeName is the type name from the spec file
+ * def is the definition of the type. e.g. the class declaration of the enum class
+ * type is a loosely defined tag used by generator e.g. enum, array, etc.
+ * propertyName is the class/struct member name
+ * goPropertyName, is when provided with x-go-name
+ * doc is the comments associated with the parameter
+ * required indeicates if the field is a required field
+ * e.g. For enum type, typeName will be enum class name, def will be the enum
+ */
+public class TypeDef {
+    public TypeDef(
+            String javaTypeName, 
+            String rawTypeName,
+            String def, 
+            String type, 
+            String propertyName,
+            String goPropertyName,
+            String doc,
+            boolean required) {
+        this.javaTypeName = javaTypeName;
+        this.rawTypeName = rawTypeName;
+        this.def = def;
+        this.type = type;
+        this.propertyName = propertyName;
+        this.goPropertyName = goPropertyName;
+        this.doc = doc;
+        this.required = required;
+    }
+    public TypeDef(String typeName,
+            String rawTypeName,
+            String propertyName, 
+            String goPropertyName,
+            String doc,
+            boolean required) {
+        this.javaTypeName = typeName;
+        this.rawTypeName = rawTypeName;
+        this.def = null;
+        this.type = null;
+        this.propertyName = propertyName;
+        this.goPropertyName = goPropertyName;
+        this.doc = doc;
+        this.required = required;
+    }
+    public boolean isOfType(String type) {
+        if (this.type == null) {
+            return false;
+        }
+        return this.type.contains(type);
+    }
+
+    @Override
+    public String toString() {
+        throw new RuntimeException("Should not get the string value of the object directly!");
+    }
+
+    public String javaTypeName;
```

Comment:
Would this be factored out if we converted the java generator to use the publish/subscribe system?

Reply:
Yes. This and a lot more may need to change for that purpose.
Maybe after the release?